tests/testthat/test-brreg.R

```diff
@@ -0,0 +1,123 @@
+entity_resp <- readRDS('../testdata/brreg-entity-resp.rds')
+# skip_if_offline(host = '')
+
+# ---- get_entity() ----
+
+test_that('get_entity() works for single id queries', {
+
+  # --- Numeric input ---
+  df <- get_entity(974760843)
+  # Check type
+  expect_equal(class(df), 'data.frame')
+  # Check rows
+  expect_equal(nrow(df), 1)
+  expect_true(df$navn == 'RIKSREVISJONEN')
+  # Check column names
+  expect_true(all(names(df) == names(entity_resp)))
+  # Check column classes
+  expect_true(all(sapply(df, class) == sapply(entity_resp, class)))
+
+  # --- Character input ---
+  df <- get_entity('974760843')
+  # Check rows
+  expect_equal(nrow(df), 1)
+  expect_true(df$navn == 'RIKSREVISJONEN')
+  # Check column names
+  expect_true(all(names(df) == names(entity_resp)))
+  # Check column classes
+  expect_true(all(sapply(df, class) == sapply(entity_resp, class)))
+
+})
+
+test_that('get_entity() works for multitiple id queries', {
+  df <- get_entity(c(974760843, 971524960))
+  # Check rows
+  expect_equal(nrow(df), 2)
+  expect_true(all(df$navn == c('RIKSREVISJONEN', 'STORTINGET')))
+  # Check column names
+  expect_true(all(names(df) == names(entity_resp)))
+  # Check column classes
+  expect_true(all(sapply(df, class) == sapply(entity_resp, class)))
+})
+
+test_that('get_entity() works for single names queries', {
+  df <- get_entity('Riksrevisjonen')
+  # Check rows
+  expect_gte(nrow(df), 1)
+  expect_true(all(grepl('RIKSREVISJONEN', df$navn)))
+  # Check column names
+  expect_true(all(names(df) == names(entity_resp)))
+  # Check column classes
+  expect_true(all(sapply(df, class) == sapply(entity_resp, class)))
+})
+
+test_that('get_entity() works for multitiple name queries', {
+  df <- get_entity(c('Riksrevisjonen', 'Stortinget'))
+  # Check rows
+  expect_gte(nrow(df), 2)
+  expect_true(all(grepl('RIKSREVISJONEN|STORTINGET', df$navn)))
+  # Check column names
+  expect_true(all(names(df) == names(entity_resp)))
+  # Check column classes
+  expect_true(all(sapply(df, class) == sapply(entity_resp, class)))
+})
+
+test_that('get_entity() works when raw_response = TRUE', {
+  res <- get_entity(974760843, raw_response = TRUE)
+  expect_true(is.list(res))
+  expect_true(is.list(res[[1]]))
+  expect_equal(names(res[[1]]), c('url', 'status', 'content', 'response'))
+  expect_identical(class(res[[1]]), "noAPI")
+  expect_true(is.list(res[[1]]$content))
+})
+
+test_that('get_entity() returns NULL on failures', {
```

Comment:
If mod11 check is implemented, these tests should be rewritten

Reply:
Tests have been updated. Note however that since `get_entity()` is serialised it is helpful for the function to just print the error, but return NULL.  The internal function `get_entity_single()`  should however fail, hence the use of `expect_error()` for the similar tests for that function.